### PR TITLE
Add a port of libotf

### PR DESCRIPTION
**Pull request by sappigeninja**
Ported libotf to managarm.
Had some issues, so might have pushed a few times.

### CODE REVIEW

File: bootstrap.d/dev-libs.yml [hunk 1]
```diff
@@ -526,6 +526,42 @@ packages:
           DESTDIR: '@THIS_COLLECT_DIR@'
         quiet: true
 
+  - name: libotf
+    source:
+      subdir: 'ports'
+      git: 'https://salsa.debian.org/debian/libotf.git'
+      patch-path-strip: 1
```
Comment (Dennisbonke) [inline bootstrap.d/dev-libs.yml:533]: With git ports, this line is not needed.

**Comment by Dennisbonke** [comment]
LGTM but can you squash the commits into one commit?

**Comment by sappigeninja** [comment]
> LGTM but can you squash the commits into one commit?

Squashed it into one.
And removed the subdir part in the libotf entry.

**Comment by Dennisbonke** [comment]
Can you remove the `patch-path-strip` line and the commented out lines? They serve no purpose here.

**Comment by Dennisbonke** [comment]
Does not build for me with the following error:

```
dennis@DENNIS-MAIN-LAPTOP-SIDUCTION:~/managarm-clone/build$ xbstrap install --recon libotf
xbstrap: Running the following plan:
    fetch          libotf
    checkout       libotf
    patch          libotf
    regenerate     libotf
    configure      libotf
    build          libotf
    install        libotf
xbstrap: fetch libotf [1/7]
Initialized empty Git repository in /mnt/linux/managarm/managarm-clone/src/libotf/.git/
remote: Enumerating objects: 396, done.
remote: Counting objects: 100% (396/396), done.
remote: Compressing objects: 100% (129/129), done.
remote: Total 396 (delta 265), reused 377 (delta 255), pack-reused 0
Receiving objects: 100% (396/396), 825.17 KiB | 1.17 MiB/s, done.
Resolving deltas: 100% (265/265), done.
From https://salsa.debian.org/debian/libotf
 * [new branch]      master     -> origin/master
xbstrap: checkout libotf [2/7]
Reset branch 'master'
Branch 'master' set up to track remote branch 'master' from 'origin'.
xbstrap: patch libotf [3/7]
Patch format detection failed.
xbstrap: Action patch of source libotf failed
dennis@DENNIS-MAIN-LAPTOP-SIDUCTION:~/managarm-clone/build$
```

The patch files don't look like a typical `git format-patch` to me, you may need to recheck that.

**Comment by sappigeninja** [comment]
I had the wrong patches, fixed it now.

File: bootstrap.d/dev-libs.yml [hunk 2]
```diff
@@ -526,6 +526,40 @@ packages:
...
+      tools_required:
+        - host-autoconf-v2.69
+        - host-automake-v1.15
+        - host-libtool
+        - host-pkg-config
+      regenerate:
+        - args: ['./autogen.sh']
+        - args: ['cp',
+            '@BUILD_ROOT@/tools/host-automake-v1.15/share/automake-1.15/config.sub',
+            '@THIS_SOURCE_DIR@/']
```
Comment (avdgrinten) [inline bootstrap.d/dev-libs.yml:544]: autogen doesn't regenerate this automatically?

File: bootstrap.d/dev-libs.yml [hunk 3]
```diff
@@ -526,6 +526,40 @@ packages:
           DESTDIR: '@THIS_COLLECT_DIR@'
         quiet: true
 
+  - name: libotf
+    source:
+      git: 'https://salsa.debian.org/debian/libotf.git'
```
Comment (avdgrinten) [inline bootstrap.d/dev-libs.yml:531]: This does not seem to be the upstream repo (?) but I also could not find the upstream one.

Comment (avdgrinten) [reply bootstrap.d/dev-libs.yml:531]: Where do other distros like Void or Gentoo build from?

Comment (Dennisbonke) [reply bootstrap.d/dev-libs.yml:531]: Also, I'm missing `subdir: 'ports'` here.

File: bootstrap.d/dev-libs.yml (hunk 2 above)
Comment (sappigeninja) [reply bootstrap.d/dev-libs.yml:544]: Nope it couldnt find managarm as the host.

File: bootstrap.d/dev-libs.yml (hunk 3 above)
Comment (sappigeninja) [reply bootstrap.d/dev-libs.yml:531]: > Where do other distros like Void or Gentoo build from?

Gentoo builds from the official website of the project. But that is only a tarball. So i opted for a github mirror as i heard that it is preferred.

Comment (sappigeninja) [reply bootstrap.d/dev-libs.yml:532]: > Also, I'm missing `subdir: 'ports'` here.

Fixed it

Comment (Dennisbonke) [reply bootstrap.d/dev-libs.yml:532]: > 
> 
> > Where do other distros like Void or Gentoo build from?
> 
> 
> 
> Gentoo builds from the official website of the project. But that is only a tarball. So i opted for a github mirror as i heard that it is preferred.
> 
> 

Regarding Void, they also build from a tarball. As this seems to be the Debian mirror, I’d say that is good enough.

File: bootstrap.d/dev-libs.yml [hunk 4]
```diff
@@ -526,6 +526,41 @@ packages:
           DESTDIR: '@THIS_COLLECT_DIR@'
         quiet: true
 
+  - name: libotf
+    subdir: ports
```
Comment (Dennisbonke) [inline bootstrap.d/dev-libs.yml:530]: This line should be below the `source:` declaration I think

**Review by Dennisbonke — CHANGES_REQUESTED** [review]

**Comment by Dennisbonke** [comment]
LGTM now, thanks for the PR!

**Pull request merged**